packages/core/mesh/teleport/src/muxing/framer.ts

```diff
@@ -50,11 +62,19 @@ export class Framer {
     send: (message) => {
       // log('write', { len: message.length, frame: Buffer.from(message).toString('hex') })
       return new Promise<void>((resolve) => {
-        const canContinue = this._stream.push(encodeFrame(message));
-        if (!canContinue) {
-          this._responseQueue.push(resolve);
-        } else {
-          process.nextTick(resolve);
+        const isEmpty = this._chunkQueue.length === 0;
```

Comment:
Is this happening after the balancer? I think it would be better to split up the chunks somewhere before or at the balancer level, so that one single big chunk doesnt block the entire stream

Reply:
i was thinking about that as well; would it require adding some "envelope" to each chunk, and adding a way of bringing those chunks back together to a single message?
(It seems like currently framer would not work if chunks from different streams will be mixed together)
was wondering how much of overheat that could cause